packages/ui/components/ui/select-account.tsx

```diff
@@ -1,79 +1,152 @@
-import { ArrowLeftIcon, ArrowRightIcon } from 'lucide-react';
+import { ArrowLeftIcon, ArrowRightIcon, CopyIcon, InfoIcon } from 'lucide-react';
 import { Input } from './input';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Button } from './button';
 import { cn } from '../../lib/utils';
+import { Identicon } from './identicon';
+import { CopyToClipboard } from './copy-to-clipboard';
+import { Switch } from './switch';
+import { IncognitoIcon } from './icons/incognito';
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from './tooltip';
+import { Account } from '@penumbra-zone/types';
 
 interface SelectAccountProps {
-  index: number | undefined;
-  previous: () => void;
-  next: () => void;
-  setIndex: (index: number) => void;
+  getAccount: (index: number, ephemeral: boolean) => Promise<Account> | Account | undefined;
 }
 const MAX_INDEX = 2 ** 32;
 
-export const SelectAccount = ({ index, previous, next, setIndex }: SelectAccountProps) => {
-  const [width, setWidth] = useState(index?.toString().length ?? 1);
+export const SelectAccount = ({ getAccount }: SelectAccountProps) => {
+  const [index, setIndex] = useState<number>(0);
+  const [ephemeral, setEphemeral] = useState<boolean>(false);
 
-  return (
-    <div className='flex items-center justify-between'>
-      <Button variant='ghost' className={cn('hover:bg-inherit', index === 0 && 'cursor-default')}>
-        {index !== 0 ? (
-          <ArrowLeftIcon
-            onClick={() => {
-              previous();
+  const [width, setWidth] = useState(index.toString().length);
+  const [account, setAccount] = useState<Account | undefined>();
 
-              setWidth(Number(String(index! - 1).length));
-            }}
-            className='h-6 w-6 hover:cursor-pointer'
-          />
-        ) : (
-          <span className='h-6 w-6' />
-        )}
-      </Button>
-      <div className='select-none text-center font-headline text-xl font-semibold leading-[30px]'>
-        {index !== undefined && (
-          <div className='flex items-end gap-[6px]'>
-            <span>Account</span>
-            <div className='flex items-end gap-0'>
-              <p>#</p>
-              <div className='relative w-min min-w-[24px]'>
-                <Input
-                  variant='transparent'
-                  type='number'
-                  className='mb-[3px] h-6 py-[2px] font-headline text-xl font-semibold leading-[30px]'
-                  onChange={e => {
-                    const value = Number(e.target.value);
-                    const valueLength = e.target.value.replace(/^0+/, '').length;
+  useEffect(() => {
+    void (async () => {
+      const account = await getAccount(index, ephemeral);
+      setAccount(account);
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why not put `getAccount` as a dep? Also, we shouldn't have eslint-disable comments without an additional comment on why it's there. Please add code comment to why it's necessary.

Reply:
this updates the ephemeral address of each block